modules/aws/vendor-access/main.tf

```diff
@@ -32,12 +32,13 @@ locals {
   allowed_iam_policies       = join(", ", formatlist("\"%s\"", distinct(concat(local.additional_iam_policy_arns, local.default_allowed_iam_policies))))
   arn_like_vpcs              = formatlist("\"arn:%s:ec2:%s:%s:vpc/%s\"", local.aws_partition, var.region, local.account_id, var.vpc_allowed_ids)
   arn_like_vpcs_str          = format("[%s]", join(",", local.arn_like_vpcs))
-  assume_conditions          = concat(local.external_id, local.source_identity, local.principal_check, local.vendor_federation)
-  support_assume_conditions  = concat(local.external_id, local.source_identity)
+  assume_conditions          = length(var.external_ids) != 0 ? concat(local.external_ids, local.source_identity, local.principal_check, local.vendor_federation) : concat(local.external_id, local.source_identity, local.principal_check, local.vendor_federation)
```

Comment:
We should append the external_id to the external_ids, so existing user won't be break.

Reply:
If the external_ids is an empty array it will default to the existing behaviour, so the TF will show no changes. 
If we append the external_id to the end of the array does that mean we also want to update the policy from `StringEquals` to `ForAllValues:StringEquals`?